doc/tdshim_spec.md

```diff
@@ -762,6 +762,74 @@ The expected flow is:
    TDH.MR.EXTEND as described above, from the lowest GPA to the highest GPA,
    with 256 bytes granularity.
 
+#### Guideline
+
+TD Shim should follow [TCG Platform Firmware Profile specification](#tcg-pfp)
+in general.
+
+1. About separator
+
+   TD Shim shall measure 0x00000000 or 0xFFFFFFFF to RTMR[0~1] and the matching
+   EV_SEPARATOR event shall be recorded in the event log, prior to the
+   invocation of TD payload. As such, the verifier can know the rest events are
+   for the TD payload.
+
+   In case of an error occurs, TD shim shall measure 0x00000001 to RTMR[0~1] and
+   the matching EV_SEPARATOR event shall be recorded in the event log.
+
+2. About TD configuration
+
+   TD Shim shall extend the TD Hob to RTMR[0] and the event should be
+   EV_PLATFORM_CONFIG_FLAGS with event data being TD_SHIM_PLATFORM_CONFIG_INFO
+   and Descriptor being "td_hob\0", Info being the whole TD Hob.
+
+3. About payload info
+
+   TD Shim shall extend the payload binary to RTMR[1] and the event should be
+   EV_EFI_PLATFORM_FIRMWARE_BLOB2 with event data being
+   UEFI_PLATFORM_FIRMWARE_BLOB2 and BlobDescription being "td_payload\0".
+
+   TD Shim shall extend the payload parameter to RTMR[1] and the event should be
+   EV_PLATFORM_CONFIG_FLAGS with event data being TD_SHIM_PLATFORM_CONFIG_INFO
+   and Descriptor being "td_payload_info\0", Info being the payload parameter.
+
+4. About secure boot
+
+   TD Shim shall extend the security policy database (provisioned trust anchors)
+   in RTMR[0] and the event should be EV_PLATFORM_CONFIG_FLAGS with
+   event data being TD_SHIM_PLATFORM_CONFIG_INFO and Descriptor being
+   "secure_policy_db", Info being the whole security policy database.
+
+   TD Shim shall extend the security authority entry in the security policy
+   database (matched trust anchor) in RTMR[0] and the event should be
+   EV_PLATFORM_CONFIG_FLAGS with event data being
+   TD_SHIM_PLATFORM_CONFIG_INFO and Descriptor being "secure_authority", Info
+   being the matched trust anchor in the security policy database.
+
+   If the payload includes secure version number (SVN) information, then TD Shim
+   should extend the SVN of the payload to RTMR[1] and the event should be
+   EV_PLATFORM_CONFIG_FLAGS with event data being
+   TD_SHIM_PLATFORM_CONFIG_INFO and Descriptor being "td_payload_svn\0", Info
+   being a UINT64 SVN.
+
+   Please refer the data structure definition in [secure_boot](secure_boot.md).
+
+5. About no action event
+
+   TD Shim shall include the TCG_EfiSpecIDEvent as the first event and the
+   vendorInfo should be "td_shim".
+
+   TD Shim may include TCG_Sp800_155_PlatformId_Event2 to provide reference
+   manifest information.
+
+**Table 3.5-4 TD_SHIM_PLATFORM_CONFIG_INFO**
+
+| Field              | Byte Length | Byte Offset | Description                                 |
```

Comment:
Nit: It would be slightly clearer (atleast to me) if the "Byte Offset" column came before the "Byte Length" column.

Reply:
I notice that different standards are using different style, for example:
* UEFI/UEFI is using Mnemonic/Offset/Length/Description.
* UEFI/ACPI is using Field/Length/Offset/Description.
* TCG/PFP is using Type/Name/Description.
* TCG/MOR is using Offset/Size/Name/Description.
* DMTF/SPDM is using Offset/Field/Size/Description
* DMTF/SMBIOS is using Offset/Version/Name/Length/Value/Description

I will choose the style similar to existing standard.
Since we are defining something similar to TCG/PFP. I will use - Type/Name/Description.